### PR TITLE
Feature/single post pages

### LINKED ISSUE #26: Single Post Page
Back to posts and apply buttons functionalities
Make the functionality for the similar posts
Single post page for the receiver

### PR DESCRIPTION
**Pull request by betulerkoc**
- Create SinglePostHeader and SinglePostContent components

- Create SinglePostPage page and render component according to data.

single-post

### CODE REVIEW

File: src/components/SinglePostContent/index.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import "./index.css";
+import { Container, Row, Col } from "react-bootstrap";
+
+export default function SinglePostContent({ postContentData }) {
+  console.log(postContentData);
+  const targetGroup = () => {
+    return (
+      <>
+        <h2>Target Group</h2>
```
Comment (Ammar-64) [inline src/components/SinglePostContent/index.jsx]: All texts should be coming from i18next

File: src/components/SinglePostHeader/index.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+export default function SinglePostHeader({ postHeaderData }) {
+  return (
+    <Container className="postHeader">
+      <Row className="headerSection">
+        <Col lg={10}>
+          <h2 className="postTitle">{postHeaderData.title}</h2>
+          <h6>{postHeaderData.description}</h6>
+        </Col>
+        <Col lg={2}>
+          <Button className="backButton" variant="light">
```
Comment (Ammar-64) [inline src/components/SinglePostHeader/index.jsx]: Texts should be coming from i18next

**Review by Ammar-64 — CHANGES_REQUESTED** [review]
Solid work, few comments and we’re good to go but can you delete howItWorks component, it was an example and it unnecessary anymore

**Comment by Ammar-64** [comment]
@betulerkoc there’s a conflict with dev, can you please fix it

**Comment by Ammar-64** [comment]
@betulerkoc the post pages are merged now, so you can pull and connect them together

File: src/components/SinglePostHeader/index.jsx (hunk 2 above)
Comment (betulerkoc) [reply src/components/SinglePostHeader/index.jsx]: PTAL

**Review by knargocer — APPROVED** [review]

File: src/components/Post/index.jsx [hunk 3]
```diff
@@ -15,15 +16,19 @@ export default function PostVolunteer(postInfo) {
...
-        <p className="cardTitle text-center">{postInfo.postInfo.title}</p>
-        <p className="postDate">Date: {postInfo.postInfo.date}</p>
-        <p className="postName">By : {postInfo.postInfo.profileName}</p>
-        <p className="cardBody text-center">{postInfo.postInfo.description}</p>
-        <img
-          className="cardImg"
-          src={postInfo.postInfo.image}
-          alt={t("postPage.altVolunteerHeader")}
-        />
+        <Link to={"posts/" + postInfo.postInfo.id}>
```
Comment (Ammar-64) [inline src/components/Post/index.jsx:19]: It’s more convenient to write the link this way,  `<Link to=`posts/${postInfo.postInfo.id}` >`

File: src/components/SimilarPosts/index.jsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { Row, Container } from "react-bootstrap";
+import Post from "../Post/index";
+import "./index.css";
+
+export default function SimilarPosts({ similarPostData }) {
+  console.log(similarPostData);
+  return (
+    <Container className="similarPostContent">
+      <h1 className="text-center title">Similar Posts</h1>
```
Comment (Ammar-64) [inline src/components/SimilarPosts/index.jsx]: The text here should be coming from i18next, and I don’t think this should be h1

File: src/components/SimilarPosts/index.jsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      <Row className="justify-content-md-center">
+        {similarPostData
```
Comment (Ammar-64) [inline src/components/SimilarPosts/index.jsx]: You can use slice here like this
`similarPostData
           .slice(0,3)
           .map((data) => {
             return <Post postInfo={data} />;
           }) `

File: src/containers/SinglePostPage/index.css [hunk 6]
```diff
@@ -0,0 +1,2 @@
+.singlePost {
```
Comment (Ammar-64) [inline src/containers/SinglePostPage/index.css]: Is this necessary?

File: src/containers/SinglePostPage/index.jsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+import SinglePostHeader from "../../components/SinglePostHeader";
+import SinglePostContent from "../../components/SinglePostContent";
+import SimilarPosts from "../../components/SimilarPosts";
+import { useParams } from "react-router-dom";
+
+export default function SinglePostPage() {
+  const [currentPost, setCurrentPost] = useState({});
+  const [similarPost, setSimilarPost] = useState([]);
+  const { id } = useParams();
+  console.log(id);
```
Comment (Ammar-64) [inline src/containers/SinglePostPage/index.jsx]: No console.log should be in production

File: src/components/SimilarPosts/index.jsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { Row, Container } from "react-bootstrap";
+import Post from "../Post/index";
+import "./index.css";
+
+export default function SimilarPosts({ similarPostData }) {
+  console.log(similarPostData);
```
Comment (Ammar-64) [inline src/components/SimilarPosts/index.jsx]: No console.log should be in production

**Review by Ammar-64 — CHANGES_REQUESTED** [review]
Solid, some few things and we’re good to go

File: src/components/SimilarPosts/index.jsx (hunk 8 above)
Comment (betulerkoc) [reply src/components/SimilarPosts/index.jsx]: PTAL

File: src/containers/SinglePostPage/index.css (hunk 6 above)
Comment (betulerkoc) [reply src/containers/SinglePostPage/index.css]: PTAL

**Review by Ammar-64 — APPROVED** [review]

**Pull request merged**